src/TrackGenerator.cpp

```diff
@@ -1131,104 +1092,22 @@ void TrackGenerator::initializeTrackCycleIndices(boundaryType bc) {
 
 /**
  * @brief Set the Track UIDs for all tracks and generate the 1D array of
- *        track pointers that separates the groups of tracks by parallel group.
- * @details The Solver requires the tracks to be separated into groups of tracks
- *          that can be looped over in parallel without data races. This method
- *          creates a 1D array of Track pointers where the tracks are arranged
- *          by parallel group. If the geometry has a periodic BC, 6 periodic
- *          groups are created; otherwise, 2 periodic groups are created. The
- *          Track UIDs are also set to their index in the tracks by periodic
- *          group array.
+ *        track pointers.
  */
 void TrackGenerator::initializeTrackUids() {
 
-  Track* track;
-  bool periodic;
-  int uid = 0;
-  int azim_group_id, periodic_group_id;
-  int track_azim_group_id, track_periodic_group_id;
-  int track_periodic_index;
-  int num_tracks;
-
-  /* If periodic boundary conditions are present, set the number of parallel
-   * track groups to 6; else, set the number of parallel track groups to 2. Also
-   * set the periodic boolean indicating whether periodic bcs are present. */
-  if (_geometry->getMinXBoundaryType() == PERIODIC ||
-      _geometry->getMinYBoundaryType() == PERIODIC) {
-    _num_parallel_track_groups = 6;
-    periodic = true;
-  }
-  else {
-    _num_parallel_track_groups = 2;
-    periodic = false;
-  }
-
-  /* Allocate memory for the num tracks by parallel group array */
-  _num_tracks_by_parallel_group = new int[_num_parallel_track_groups];
-
-  /* Allocate memory for the tracks by parallel group array */
-  _tracks_by_parallel_group = new Track*[getNumTracks()];
-
-  /* Loop over the parallel track groups */
-  for (int g = 0; g < _num_parallel_track_groups; g++) {
-
-    /* Initialize the number of tracks counter */
-    num_tracks = 0;
-
-    /* Set the azimuthal and periodic group ids */
-    azim_group_id = g % 2;
-    periodic_group_id = g / 2;
-
-    /* Loop over all tracks and add all tracks belonging to the current
-     * parallel group to the tracks by parallel groups array */
-    for (int a=0; a < _num_azim_2; a++) {
-      for (int i=0; i < _num_tracks[a]; i++) {
-
-        /* Get current track */
-        track = &_tracks[a][i];
-
-        /* Get the track azim group id */
-        track_azim_group_id = a / (_num_azim_2 / 2);
-
-        /* Get the track periodic group id */
-        if (periodic) {
-          track_periodic_index = track->getPeriodicTrackIndex();
+  /* Allocate memory for the 1D tracks array */
+  _tracks_array = new Track*[getNumTracks()];
```

Comment:
Yeah, but I generate tracks 100s of times for different clustered
geometries. Should only be a two line change to deallocate if necessary.
On May 31, 2016 1:23 PM, "Geoffrey Gunow" <notifications@github.com> wrote:

> In src/TrackGenerator.cpp
> <https://github.com/mit-crpg/OpenMOC/pull/313#discussion_r65225837>:
>
> > -    /* Loop over all tracks and add all tracks belonging to the current
> > -     * parallel group to the tracks by parallel groups array */
> > -    for (int a=0; a < _num_azim_2; a++) {
> > -      for (int i=0; i < _num_tracks[a]; i++) {
> > -
> > -        /* Get current track */
> > -        track = &_tracks[a][i];
> > -
> > -        /* Get the track azim group id */
> > -        track_azim_group_id = a / (_num_azim_2 / 2);
> > -
> > -        /* Get the track periodic group id */
> > -        if (periodic) {
> > -          track_periodic_index = track->getPeriodicTrackIndex();
> > +  /* Allocate memory for the 1D tracks array */
> > +  _tracks_array = new Track*[getNumTracks()];
>
> It shouldn't be. This function is only called from within
> TrackGenerator::generateTracks.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mit-crpg/OpenMOC/pull/313/files/e6bae44ad4167266dbdc2a911da10adf8061341e#r65225837>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMyVJaelF_K77Zepsvrjos0X4jxFNAaks5qHG6hgaJpZM4IqE28>
> .
>